File: react/src/app/dto/RegistrationDto.ts
export default class RegistrationDto {

    private email: string;
    private first: string;
    private last: string;
    private password: string;
    
    constructor(email: string, first: string, last: string, password: string){
        this.email = email;
        this.first = first;
        this.last = last;
        this.password = password;
    }

    toJSON() {
        return {
            "email": this.email,
            "first": this.first,
            "last": this.last,
            "password": this.password,
        }
    }

}

File: spring/game/src/main/java/com/gigilife/game/user/dto/RegistrationDto.java
package com.gigilife.game.user.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


/*
 * 
 * Registration DTO for new users
 * /api/user/register endpoint
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RegistrationDto {

    private String email;

    private String first;

    private String last;

    private String password;
    
}
